storage/src/checker/range_set.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use btree_range_map::{AnyRange, AsRange, RangeSet};
+use std::num::NonZeroU64;
+use std::ops::Bound;
+
+use crate::nodestore::NodeStoreHeader;
+use crate::{CheckerError, LinearAddress};
+
+pub(super) struct LinearAddressRangeSet {
+    range_set: RangeSet<u64>,
```

Comment:
Would be great if we could use a`RangeSet<LinearAddress>` here. That would save some error checks later and make insertion easier since all addresses are essentially NonZeroU64s.

One snag is you might have to implement the `Measure` trait on LinearAddress/NonZerU64. I bet the crate owner would take a PR to add the non-zero types to the list of those that implement `Measure`...

Reply:
https://github.com/timothee-haudebourg/range-traits/pull/5